diff --git a/pkg/api/frontendsettings.go b/pkg/api/frontendsettings.go
index 2265726..fb818d1 100644
--- a/pkg/api/frontendsettings.go
+++ b/pkg/api/frontendsettings.go
@@ -588,7 +588,7 @@ func (hs *HTTPServer) getFSDataSources(c *contextmodel.ReqContext, availablePlug
 		// Update `jsonData.database` for outdated provisioned SQL datasources created WITHOUT the `jsonData` object in their configuration.
 		// In these cases, the `Database` value is defined (if at all) on the root level of the provisioning config object.
 		// This is done for easier warning/error checking on the front end.
-		if (ds.Type == datasources.DS_MSSQL) || (ds.Type == datasources.DS_MYSQL) || (ds.Type == datasources.DS_POSTGRES) {
+		if (dsDTO.Type == datasources.DS_MSSQL) || (dsDTO.Type == datasources.DS_MYSQL) || (dsDTO.Type == datasources.DS_POSTGRES) {
 			// Only update if the value isn't already assigned.
 			if dsDTO.JSONData["database"] == nil || dsDTO.JSONData["database"] == "" {
 				dsDTO.JSONData["database"] = ds.Database
